src/ortho.rs

```diff
@@ -72,12 +72,20 @@ impl Ortho {
         value: usize,
         version: usize,
     ) -> Vec<Ortho> {
+        // First, insert the value into the old ortho at the correct position
+        let mut old_payload_with_value = ortho.payload.clone();
+        let insert_pos = old_payload_with_value.iter().position(|x| x.is_none()).unwrap_or(old_payload_with_value.len());
+        if insert_pos < old_payload_with_value.len() {
```

Comment:
This check should not be necessary - look at expansion logic and the context surrounding this call.